### PR TITLE
Rustup to https://github.com/rust-lang/rust/pull/66789

### PR DESCRIPTION
**Pull request by phansch**
changelog: none

### CODE REVIEW

**Comment by phansch** [comment]
maybe r? @oli-obk (this PR resulted from moving `mir::SourceScopeLocalData` to `SourceScopeData` in https://github.com/rust-lang/rust/pull/66789)

File: clippy_lints/src/redundant_clone.rs [hunk 1]
```diff
@@ -208,8 +208,9 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for RedundantClone {
 
             if !used_later {
                 let span = terminator.source_info.span;
-                let node = if let mir::ClearCrossCrate::Set(scope_local_data) = &mir.source_scope_local_data {
-                    scope_local_data[terminator.source_info.scope].lint_root
+                let scope = terminator.source_info.scope;
+                let node = if let mir::ClearCrossCrate::Set(scope_local_data) = &mir.source_scopes[scope].local_data {
+                    scope_local_data.lint_root
                 } else {
                     unreachable!()
```
Comment (eddyb) [inline clippy_lints/src/redundant_clone.rs]: If you're going to panic, please use `.as_ref().assert_crate_local()` like I did in my PR.

Comment (phansch) [reply clippy_lints/src/redundant_clone.rs]: That makes sense, thanks!

**Pull request merged**